service/devdoc/registry_requirements.md

```diff
@@ -370,17 +370,234 @@ Request-Id: <guid>
+
+**SRS_NODE_IOTHUB_REGISTRY_18_005: [** The `updateModuleTwin` method shall construct an HTTP request using information supplied by the caller, as follows:
+```
+PATCH /twins/<encodeURIComponent(deviceId)/modules/<encodeURIComponent(moduleId)>>?api-version=<version> HTTP/1.1
+Authorization: <config.sharedAccessSignature>
+Content-Type: application/json; charset=utf-8
+Request-Id: <guid>
+If-Match: <etag>
+<patch>
+```
+**]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_006: [** The `updateModuleTwin` method shall call the `done` callback with a `Twin` object updated with the latest property values stored in the IoT Hub service. **]**
+
+## Device configuration
+
+### addConfiguration(configuration: Configuration, done: Registry.ResponseCallback): void;
+
+**SRS_NODE_IOTHUB_REGISTRY_18_007: [** The `addConfiguration` method shall throw a `ReferenceError` exception if `configuration` or `done` is falsy. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_008: [** The `addConfiguration` method shall throw an `ArgumentError` exception if `configuration.id` is falsy. **]**
+**SRS_NODE_IOTHUB_REGISTRY_18_009: [** The `addConfiguration` method shall set `configuration.schemaVersion` to the constant `currentConfigurationSchemaVersion` if it is not already set. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_010: [** The `addConfiguration` method shall construct an HTTP request using information supplied by the caller, as follows:
+```
+PUT /configurations/<encodeURIComponent(configuration.id)>?api-version=<version> HTTP/1.1
+Authorization: <sharedAccessSignature>
+Content-Type: application/json; charset=utf-8
+Request-Id: <guid>
+
+<configuration>
+```
+**]**
+
+### getConfiguration(configurationId: string, done: Registry.ResponseCallback): void;
+
+**SRS_NODE_IOTHUB_REGISTRY_18_011: [** The `getConfiguration` method shall throw a `ReferenceError` exception if `configurationId` is falsy. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_012: [** The `getConfiguration` method shall construct an HTTP request using information supplied by the caller, as follows:
+```
+GET /configurations/<encodeURIComponent(configurationId)>?api-version=<version> HTTP/1.1
+Authorization: <sharedAccessSignature>
+Request-Id: <guid>
+```
+**]**
+
+### getConfigurations(done: Registry.ResponseCallback): void;
+
+**SRS_NODE_IOTHUB_REGISTRY_18_013: [** The `getConfigurations` method shall throw a `ReferenceError` exception if `done` is falsy. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_014: [** The `getConfigurations` method shall construct an HTTP request using information supplied by the caller, as follows:
+```
+GET /configurations?api-version=<version> HTTP/1.1
+Authorization: <sharedAccessSignature>
+Request-Id: <guid>
+```
+**]**
+
+### updateConfiguration(configuration: Configuration, done: Registry.ResponseCallback): void;
+### updateConfiguration(configuration: Configuration, forceUpdate: boolean, done: Registry.ResponseCallback): void;
```

Comment:
nit: are those 2 necessary considering the last one cover both cases?

Reply:
This seems to be an ad-hoc typescript pattern for defining APIs when a middle parameter is optional.  The .ts file has all 3 signatures, so I put them here too.  

---
In reply to: [176890294](https://github.com/Azure/azure-iot-sdk-node/pull/256#discussion_r176890294) [](ancestors = 176890294)